v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/README_Load_Testing.md

```diff
@@ -0,0 +1,142 @@
+```shell
+export CLOUD_SQL_INSTANCE_ID=<my-instance>
+export PROJECT_NAME=<my-project>
+export NETWORK_PROJECT_ID=<my-network-project>
+export NETWORK_NAME=<my-network>
+
+gcloud beta sql instances patch ${CLOUD_SQL_INSTANCE_ID} \
+--project=${PROJECT_ID} \
+--network=projects/${NETWORK_PROJECT_ID}/global/networks/${NETWORK_NAME} \
+--no-assign-ip \
+--enable-google-private-path
+```
+
+3. Create tables in the instance by either connecting to it through the
+   [shell](https://cloud.google.com/sql/docs/mysql/connect-instance-cloud-shell) or through
+   [Cloud Sql Studio](https://cloud.google.com/sql/docs/mysql/manage-data-using-studio).
+
+   [**Sample MySQL Schema**](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/datastream-to-spanner/src/test/resources/DataStreamToSpannerLoadTesting/mysql-schema.sql)
+
+4. Create a schema file that contains a JSON template for the generated data. For more information, see the
+   [json-data-generator documentation](https://github.com/vincentrussell/json-data-generator/blob/master/README.md). Sample
+   can be found [here](https://github.com/vincentrussell/json-data-generator/blob/master/README.md). Upload the schema file
+   to a Cloud Storage bucket.
+
+   [**Sample Schema File**](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/datastream-to-spanner/src/test/resources/DataStreamToSpannerLoadTesting/datagenerator-schema.json)
+
+5. Run the template by following this [guide](https://cloud.google.com/dataflow/docs/guides/templates/provided/streaming-data-generator)
+```shell
+export PROJECT=<my-project>
+export BUCKET_NAME=<bucket-name>
+export REGION=<region>
+export TEMPLATE_SPEC_GCSPATH="gs://dataflow-templates-${REGION}/latest/flex/Streaming_Data_Generator"
+export QPS=<qps>
+export MESSAGE_LIMIT=<row-count>
+export SCHEMA_LOCATION=<schemaLocation>
+export DRIVER_CLASS_NAME=<driverClassName>
+export CONNECTION_URL=<connectionUrl>
+export USERNAME=<username>
+export PASSWORD=<password>
+export STATEMENT=<statement>
+export NETWORK_NAME=<my-network>
+export SUBNETWORK=<my-subnetwork>
+
+gcloud dataflow flex-template run streaming-data-generator-job --project ${PROJECT}   \
+--region ${REGION}   --template-file-gcs-location $TEMPLATE_SPEC_GCSPATH  \
+--parameters ^~^qps=${QPS}~messagesLimit=${MESSAGE_LIMIT}~schemaLocation=${SCHEMA_LOCATION}~sinkType=JDBC~driverClassName=${DRIVER_CLASS_NAME}~connectionUrl=${CONNECTION_URL}~username=${USERNAME}~password=${PASSWORD}~statement=${STATEMENT}~network=${NETWORK_NAME}~subnetwork=${SUBNETWORK}
+```
+
+### Data Generation using CSV Import
+This approach for data generation generates a csv file and imports it to a CLoud Sql Instance. All rows are identical
+and primary key is generated through auto-increment.
+1. Create tables in the instance with Primary Key as Auto-Increment by either connecting to it through the 
```

Comment:
We should mention this relying on autoincrement approach is recommended if they want to upload large csvs, which take very long to generate/upload.

Thats why we are doing the gsutil concat etc for faster generation.